python/sanlock.c

```diff
@@ -159,6 +159,33 @@ pystring_as_cstring(PyObject *obj)
 #endif
 }
 
+static int
+parse_single_disk(PyObject* disk, struct sanlk_disk* res_disk)
+{
+    int rv = 0;
+    PyObject *path = NULL;
+    uint64_t offset;
+
+    if (!PyTuple_Check(disk)) {
+         set_error(PyExc_ValueError, "Invalid disk %s", disk);
+         goto exit_fail;
+    }
+
+    if (!PyArg_ParseTuple(disk, "O&K", pypath_converter, &path, &offset)) {
+        /* Override the error since it confusing in this context. */
+        set_error(PyExc_ValueError, "Invalid disk %s", disk);
+        goto exit_fail;
+    }
+
+    strncpy(res_disk->path, PyBytes_AsString(path), SANLK_PATH_LEN - 1);
+    res_disk->offset = offset;
+    rv = 1;
+
+exit_fail:
+    Py_XDECREF(path);
+    return rv;
+}
+
```

Comment:
Looks great, so much simpler now.

Reply:
should be "finally" since we handle success as well